e_commerce_app/HomeView/HomeCollectionViewController.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  HomeCollectionViewController.swift
+//  e_commerce_app
+//
+//  Created by Bhavin Kapadia on 2022-10-09.
+//
+
+import UIKit
+
+private let reuseIdentifier = "ProductCell"
+
+
+class HomeCollectionViewController: UICollectionViewController {
+    
+    private let homeViewModel: HomeViewModel
+    private var productItems: [Product] = []
```

Comment:
When the project is nearly done and you don't have anything else to merge, you could move this to the view model. The view model should be in control of this and then the view model can provide the content necessary for the view controller to display. I have another comment about this below.